addons/link_tracker/models/mail_render_mixin.py

```diff
@@ -86,3 +91,60 @@ def _shorten_links_text(self, content, link_tracker_vals, blacklist=None, base_u
                 content = re.sub(re.escape(original_url) + r'(?![\w@:%.+&~#=/-])', link.short_url, content)
 
         return content
+
+    @api.model
+    def _prepare_link_tracker_conversion(self, root_node, base_url, short_schema, blacklist):
```

Comment:
Can't we make it a tool somewhere ? Not sure it belongs to the model (it essentially runs on a lxml node so seems a tool could be used here).